draft-ietf-httpbis-origin-frame.md

```diff
@@ -269,15 +292,15 @@ The set of origins advertised using this mechanism is under control of the serve
 obligated to use it, or to advertise all origins which they might be able to answer a request for.
 
 For example, it can be used to inform the client that the connection is to only be used for the
-SNI-based origin, by sending an empty ORIGIN frame. Or, a larger number of origins can be indicated
-by including a payload.
+SNI-based origin, by sending only an ORIGIN_SUPPORT setting. Or, a larger number of origins can be
+indicated by including one or more ORIGIN frames as well.
 
 Generally, this information is most useful to send before sending any part of a response that might
 initiate a new connection; for example, `Link` headers {{?RFC5988}} in a response HEADERS, or links
 in the response body.
 
-Therefore, the ORIGIN frame ought be sent as soon as possible on a connection, ideally before any
-HEADERS or PUSH_PROMISE frames.
+Therefore, the ORIGIN_SUPPORT setting and any ORIGIN frames ought be sent as soon as possible on a
+connection, ideally before any HEADERS or PUSH_PROMISE frames.
```

Comment:
I think that you want to point out that the ORIGIN_SUPPORT setting should be sent in the connection preface, and separately that ORIGIN frames need to be sent as early as possible.

Reply:
yup.

Reply:
Can the behavior of late ORIGIN frames be defined? As I currently read the spec, the Origin set can be modified in the full stream lifetime, so even after HEADERS or PUSH_PROMISE frames.

When the server was unable to prepare the Origin frame (e.g. due to validation delays), can it send it anyway as hint for future requests?